.github/workflows/pull-request.yml

```diff
@@ -0,0 +1,43 @@
+name: Build from Pull Request
+on:
+  pull_request:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: 📥 Checkout workspace-images
+        uses: actions/checkout@v2
+        with:
+          repository: gitpod-io/workspace-images
+
+      - name: 🔆 Install dazzle
+        run: |
+          curl -sSL https://github.com/gitpod-io/dazzle/releases/download/v0.1.6/dazzle_0.1.6_Linux_x86_64.tar.gz | sudo tar -xvz -C /usr/local/bin
+
+      - name: 🏗️ Setup buildkit
+        run: |
+          curl -sSL https://github.com/moby/buildkit/releases/download/v0.9.3/buildkit-v0.9.3.linux-amd64.tar.gz | sudo tar xvz -C /usr
+          sudo buildkitd --oci-worker=true --oci-worker-net=host --debug --group docker &
+          sudo su -c "while ! test -S /run/buildkit/buildkitd.sock; do sleep 0.1; done"
+          sudo chmod +777 /run/buildkit/buildkitd.sock
+
+      - name: 🗄️ Registry cache
+        uses: actions/cache@v2
+        with:
+          path: ~/registry
+          key: ${{ runner.os }}-pull-request-cache
+
+      - name: 📦 Setup local registry
+        run: |
+          docker run -it --detach --publish 5000:5000 --volume ~/registry:/var/lib/registry registry:2
+
+      - name: 🔨 Dazzle build
+        run: |
+          cd dazzle-rewrite
+          dazzle build localhost:5000/workspace-images
```

Comment:
Please add the flag `--chunked-without-hash` (@csweichel suggestion)

Reply:
@aledbf @csweichel @princerachit 

A side effect of doing two builds, is that each build takes an hour now. :thinking: [[1](https://github.com/gitpod-io/workspace-images/actions/runs/1653880874)][[2](https://github.com/gitpod-io/workspace-images/actions/runs/1653974963)]

For pull requests, I think it may be helpful to omit `--chunked-without-hash`, because then we can leverage the cache. 

For main, we need `--chunked-without-hash`. We can disable running of tests if `--chunked-without-hash` is specfieid (a dazzle code change because no flag exists), knowing that the next `dazzle build` will run the chunk tests. 

WDYT?

Reply:
>For pull requests, I think it may be helpful to omit --chunked-without-hash, because then we can leverage the cache

Sounds good to me. Let me get rid of this for PRs

>For main, we need --chunked-without-hash. We can disable running of tests if --chunked-without-hash is specfieid (a dazzle code change because no flag exists), knowing that the next dazzle build will run the chunk tests.

👍🏾

Reply:
I am resolving this conversation, feel free to comment again if you don't agree with changes